src/wallet/rpcwallet.cpp

```diff
@@ -3482,6 +3482,7 @@ UniValue z_getoperationstatus_IMPL(const UniValue& params, bool fRemoveFinishedO
 // If input notes are small, we might actually require more than one joinsplit per zaddr output.
 // For now though, we assume we use one joinsplit per zaddr output (and the second output note is change).
 // We reduce the result by 1 to ensure there is room for non-joinsplit CTransaction data.
+#define Z_SENDMANY_MAX_ZADDR_OUTPUTS_BEFORE_SAPLING    ((MAX_TX_SIZE_BEFORE_SAPLING / GetSerializeSize(JSDescription(), SER_NETWORK, PROTOCOL_VERSION)) - 1)
 #define Z_SENDMANY_MAX_ZADDR_OUTPUTS    ((MAX_TX_SIZE / GetSerializeSize(JSDescription(), SER_NETWORK, PROTOCOL_VERSION)) - 1)
```

Comment:
so if we agree with @ebfull 's comment above this should also have `AFTER_SAPLING` added?

Reply:
Still, it seems Sean's arguments that adding `AFTER_SAPLING` forces you to check that no bugs were introduced in all places is valid. And it's a small change. I wouldn't insist if it didn't relate to my second worry - the duplicate code between the three rpc methods. And I wonder if we can refactor some of it in this PR.